be/src/runtime/stream_load/stream_load_pipe.h

```diff
@@ -108,6 +110,29 @@ class StreamLoadPipe : public MessageBodySink {
     Status _err_st = Status::OK();
 };
 
+class StreamLoadPipeReader {
+public:
+    StreamLoadPipeReader(std::shared_ptr<StreamLoadPipe> pipe) : _pipe(pipe) {}
+    virtual ~StreamLoadPipeReader() {}
+    virtual StatusOr<ByteBufferPtr> read() { return _pipe->read(); }
+
+private:
+    std::shared_ptr<StreamLoadPipe> _pipe;
+};
+
+class CompressedStreamLoadPipeReader : public StreamLoadPipeReader {
+public:
+    CompressedStreamLoadPipeReader(std::shared_ptr<StreamLoadPipe> pipe, TCompressionType::type compression_type);
+    ~CompressedStreamLoadPipeReader() override = default;
+    StatusOr<ByteBufferPtr> read() override;
+
+private:
+    const size_t DEFAULT_DECOMPRESS_BUFFER_SIZE = 32 * 1024 * 1024;
```

Comment:
For all stream load, allocate one buffer of 32M?

Reply:
If StreamLoad concurrency is relatively high, there is a risk of excessive memory usage.？

Reply:
Now, the  buffer size grows exponentially